common/src/importers/onepasswordImporters/onepassword1PuxImporter.ts

```diff
@@ -0,0 +1,636 @@
+          cipher.identity.city = this.getValueOrDefault(city);
+          if (!this.isNullOrWhitespace(country)) {
+            cipher.identity.country = country.toUpperCase();
+          }
+          cipher.identity.postalCode = this.getValueOrDefault(zip);
+          cipher.identity.state = this.getValueOrDefault(state);
+          return;
+        }
+
+        switch (category) {
+          case CategoryEnum.Identity:
+            break;
+          case CategoryEnum.DriversLicense:
+            if (this.fillDriversLicense(field, fieldValue, cipher)) {
+              return;
+            }
+            break;
+          case CategoryEnum.OutdoorLicense:
+            if (this.fillOutdoorLicense(field, fieldValue, cipher)) {
+              return;
+            }
+            break;
+          case CategoryEnum.Membership:
+            if (this.fillMembership(field, fieldValue, cipher)) {
+              return;
+            }
+            break;
+          case CategoryEnum.Passport:
+            if (this.fillPassport(field, fieldValue, cipher)) {
+              return;
+            }
+            break;
+          case CategoryEnum.RewardsProgram:
+            if (this.fillRewardsProgram(field, fieldValue, cipher)) {
+              return;
+            }
+            break;
+          case CategoryEnum.SocialSecurityNumber:
+            if (this.fillSSN(field, fieldValue, cipher)) {
+              return;
+            }
+            break;
+          default:
+            break;
+        }
+      }
+
+      // Do not include a password field if it's already in the history
+      if (
+        field.title === "password" &&
+        cipher.passwordHistory != null &&
+        cipher.passwordHistory.some((h) => h.password === fieldValue)
+      ) {
+        return;
+      }
+
+      // TODO ?? If one of the fields is marked as guarded, then activate Password-Reprompt for the entire item
+      if (field.guarded && cipher.reprompt === CipherRepromptType.None) {
+        cipher.reprompt = CipherRepromptType.Password;
+      }
+
+      const fieldType = valueKey === "concealed" ? FieldType.Hidden : FieldType.Text;
+      this.processKvp(cipher, fieldName, fieldValue, fieldType);
+    });
+  }
+
+  private getFieldName(id: string, title: string): string {
+    if (this.isNullOrWhitespace(title)) {
+      return id;
+    }
+
+    // Naive approach of checking if the fields id is usable
+    if (id.length > 25 && RegExp(/[0-9]{2}[A-Z]{2}/, "i").test(id)) {
+      return title;
+    }
+    return id;
+  }
+
+  // totp?: string | null;
+  // date?: number | null;
+  // string?: string | null;
+  // concealed?: string | null;
+  // email?: string | null;
+  // phone?: string | null;
+  // menu?: string | null;
+  // gender?: string | null;
+  // monthYear?: number | null;
+  // url?: string | null;
+  // address?: Address | null;
+  // creditCardType?: string | null;
+  // creditCardNumber?: string | null;
+  // reference?: string | null;
```

Comment:
These look like the fields for the `Value` interface; do we need them commented here?

Reply:
Removed. Just used that as a reference, so I didn't have to jump between files